scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -246,6 +246,8 @@ def json_report(self, filename, version="NA", platform=None):
         for instance in self.instances.values():
             if platform and platform != instance.platform.name:
                 continue
+            if instance.status == "filtered" and not self.env.options.report_filtered:
```

Comment:
I wonder can we just use here the existent `----detailed-skipped-report` option ?

Reply:
prefer a new option specific for filtered status, skipped != filtered.

Reply:
This change removes from twister.json every "filtered" tests (quarantine, command line, skip, platform key etc.).
`--detailed-skipped-report` is used only to skip "filtered" tests from Junit report (however, junit is generated from twister.json, so that option does nothing, unless you use `--report-filtered`).